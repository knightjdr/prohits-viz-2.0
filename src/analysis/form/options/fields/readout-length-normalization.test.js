import React from 'react';
import { shallow } from 'enzyme';

import ReadoutLengthNormalization from './readout-length-normalization';

describe('ReadoutLengthNormalization', () => {
  test('Renders', () => {
    const wrapper = shallow(
      <ReadoutLengthNormalization
        analysisType="dotplot"
        options={[]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
