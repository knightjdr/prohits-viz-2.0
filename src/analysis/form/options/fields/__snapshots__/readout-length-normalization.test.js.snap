// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadoutLengthNormalization Renders 1`] = `
<div
  className="form__option-cs"
>
  <div
    className="form__option-cs-checkbox"
  >
    <div
      className="form__option-cs-checkbox-label"
    >
      Normalize to readout length:
    </div>
    <CustomField
      name="readoutLengthNorm"
      onChange={[Function]}
      type="switch"
    />
  </div>
  <div
    className="form__option-cs-select"
  >
    <CustomField
      helpMessage={
        <div>
          The abundance value of each readout can be adjusted to it length (gene/protein length) if a column with length is available in the input file. This option should be selected if you are worried the length may be unduly affecting the observed abundance. The adjusted value of a readout is calculated by dividing the median length of all readouts in the data set by the length of readout i and then multiplying by the abundance of readout i.
          <div
            className="form__info-img form__info-img_padding"
          >
            <img
              alt="Readout length normalization"
              src="length-normalization.svg"
            />
          </div>
          If you are applying multiple transformations to your data, readout length normalization will always occur after control subtraction but before condition normalization and log transformation.
        </div>
      }
      label="Readout length column"
      name="readoutLength"
      onChange={[Function]}
      options={Array []}
      placeHolder="Readout length column..."
      type="select"
    />
  </div>
</div>
`;
